casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -218,25 +221,66 @@ object BlockAPI {
     }
   }
 
-  def getBlocksResponse[F[_]: Monad: MultiParentCasperRef: Log: SafetyOracle: BlockStore]
-    : F[BlocksResponse] = {
+  def showBlocks[F[_]: Monad: MultiParentCasperRef: Log: SafetyOracle: BlockStore](
+      depth: Long
+  ): F[List[BlockInfosAtHeight]] = {
+    def casperResponse(implicit casper: MultiParentCasper[F]) =
+      for {
+        dag         <- MultiParentCasper[F].blockDag
+        startHeight = dag.topoSort.length - depth
+        indexedBlockInfosAtHeightList <- dag.topoSort.foldM(
+                                          (startHeight, List.empty[BlockInfosAtHeight])
+                                        ) {
+                                          case (
+                                              (height, blockInfosAtHeightAcc),
+                                              blockHashesAtHeight
+                                              ) =>
+                                            for {
```

Comment:
I would prefer to extract such large expressions into functions to increase readability a little bit.

Reply:
Done